test/object-store/util/sync/sync_test_utils.cpp

```diff
@@ -737,7 +737,7 @@ struct BaasFLXClientReset : public TestClientReset {
         if (m_on_post_local) {
             m_on_post_local(realm);
         }
-        wait_for_upload(*realm);
+        wait_for_download(*realm);
```

Comment:
This was relying on wait_for_upload() waiting for subscription changes after resume(). The thing we're *actually* waiting for here is for is a server roundtrip so that we receive the client reset error, which wait_for_download() does guarantee.